classes/Cart.php

```diff
@@ -766,30 +766,30 @@ public function getProducts($refresh = false, $id_product = false, $id_country =
         $sql->leftJoin('image_lang', 'il', 'il.`id_image` = image_shop.`id_image` AND il.`id_lang` = ' . (int) $this->getAssociatedLanguage()->getId());
 
         /** @var array<string, mixed> $result */
-        $result = Db::getInstance()->executeS($sql);
+        $products = Db::getInstance()->executeS($sql);
 
         // Reset the cache before the following return, or else an empty cart will add dozens of queries
         $products_ids = [];
         $pa_ids = [];
-        if ($result) {
-            foreach ($result as $key => $row) {
-                $products_ids[] = $row['id_product'];
-                $pa_ids[] = $row['id_product_attribute'];
-                $specific_price = SpecificPrice::getSpecificPrice($row['id_product'], $this->id_shop, $this->id_currency, $id_country, $this->id_shop_group, $row['cart_quantity'], $row['id_product_attribute'], $this->id_customer, $this->id);
+        if ($products) {
```

Comment:
`if` structure is useless here, because `foreach` will never loop if `$products` array is empty. Cyclomatic complexity could be reduced with no cost ;)

Reply:
Here's the catch: `$products` can be `false` (and even might be `null`)